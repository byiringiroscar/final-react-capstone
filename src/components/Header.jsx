import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import './header.css';
import menu from '../images/menu.png';
import logo from '../images/squid.png';
import { addSearched } from '../redux/games/gamesSlice';

const Header = () => {
  const dispatch = useDispatch();
  const [search, setSearch] = useState('');
  useEffect(() => {
    dispatch(addSearched(search));
  });
  const handleSearch = (e) => {
    setSearch(e.target.value);
  };
  return (
    <header>
      <img className="menu logo-mobile" src={menu} alt="menu" />
      <Link to="/">
        <div className="logo-desktop">
          <img src={logo} alt="logo-desk" />
          <h3 className="logo-text">
            Game
            <span>B</span>
            oss
          </h3>
        </div>
      </Link>
      <h3 className="title">Game Viewer</h3>
      <input value={search} onChange={handleSearch} type="text" placeholder="Filter Games" />
    </header>
  );
};

export default Header;
