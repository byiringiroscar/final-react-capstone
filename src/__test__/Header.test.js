import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../redux/store';
import Header from '../components/Header';

describe('Header', () => {
  it('Render Header correctly', () => {
    const headerComponent = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <Header />
          </Provider>
        </BrowserRouter>,
      )
      .toJSON();
    expect(headerComponent).toMatchSnapshot();
  });
});
