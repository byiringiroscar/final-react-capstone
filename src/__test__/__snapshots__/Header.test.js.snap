// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Render Header correctly 1`] = `
<header>
  <img
    alt="menu"
    className="menu logo-mobile"
    src="menu.png"
  />
  <a
    href="/"
    onClick={[Function]}
  >
    <div
      className="logo-desktop"
    >
      <img
        alt="logo-desk"
        src="squid.png"
      />
      <h3
        className="logo-text"
      >
        Game
        <span>
          B
        </span>
        oss
      </h3>
    </div>
  </a>
  <h3
    className="title"
  >
    Game Viewer
  </h3>
  <input
    onChange={[Function]}
    placeholder="Filter Games"
    type="text"
    value=""
  />
</header>
`;
